api/proto/teleport/legacy/types/types.proto

```diff
@@ -6262,6 +6262,11 @@ message PluginOktaSyncSettings {
   // as a gateway to interact with Okta for SAML login, SCIM provisioning and user
   // sync. May be missing for old Okta integration installs.
   string app_name = 8;
+
+  // DisableSyncAppGroups disables syncing of app groups from Okta.
+  // This is useful when the app groups are not needed in Teleport.
+  // and integration with Okta is only used for user sync.
+  bool disable_sync_app_groups = 9;
```

Comment:
Similar comment from me but for consistency I would name this just `sync_app_groups`. Because above we have settings `sync_users` and `sync_access_lists` already.

Reply:
@flyinghermit @r0mant 
Please note that this is  existing  proto Plugin spec (that is used by existing integration)  where we want to introduce new behavior without breaking changes. 

The flag should be negative (with `disabled_` prefix) to support backward compatibility - If we add the `sync_app_groups` flag the default value after upgrading to new teleport version will be false and this is **breaking change** because updating teleport cluster to new version should not change the Okta sync behavior 



Regarding `sync_flag_name` vs `flag_name_sync` naming convention the  `enable_user_sync` is was https://github.com/gravitational/teleport/blob/586a971703870f91a20b198f5c998877ab10a5ed/api/types/types.pb.go#L15740

where  all sunc flags in `PluginOktaSyncSettings` starts with sync_prefix:`sync_access_lists`  `sync_users` 

Does that sound resonable ?

Reply:
iiuc, you mean that once this field is released, we start controlling app group sync behavior with it. And with the `disable` prefix, it lets us intuitively use the default value, which will be false, in order to avoid stopping an already running app group sync. Right?

You can probably use enum instead of bool to avoid that scenario: 
```
message PluginOktaSettings {
...
  SyncAppGroup sync_app_groups = 9
}

enum SyncAppGroup {
  SYNC_APPGROUP_UNSPECIFIED = 0; <-- will be default, do nothing
  SYNC_APPGROUP_ENABLED = 1; <-- enable sync
  SYNC_APPGROUP_DISABLED = 2; <-- disable sync
}
```


Leaving that up to you and Roman.

Reply:
Ok bit of a bummer but backwards compatibility is more important than name consistency in this case (and probably not worth refactoring to enums really).